libnetwork/osl/interface_linux.go

```diff
@@ -393,6 +403,45 @@ func setInterfaceLinkLocalIPs(nlh *netlink.Handle, iface netlink.Link, i *Interf
 	return nil
 }
 
+func (n *Namespace) setSysctls(ifName string, sysctls []string) error {
+	for _, sc := range sysctls {
+		k, v, found := strings.Cut(sc, "=")
+		if !found {
+			return fmt.Errorf("expected sysctl '%s' to have format name=value", sc)
+		}
+		sk := strings.Split(k, ".")
+		if len(sk) != 3 {
+			return fmt.Errorf("expected sysctl '%s' to have format X.Y.Z, to map to net.X.Y.%s.Z", sc, ifName)
+		}
+		scPath := []string{"/proc/sys/net", sk[0], sk[1], ifName}
+		scPath = append(scPath, sk[2:]...)
+
+		sysPath := filepath.Join(scPath...)
+		errC := make(chan error, 1)
```

Comment:
`n.InvokeFunc` is a synchronous blocking call so there is no need to communicate through a channel.
```go
var errF error
f := func() {
    if err := ...; err != nil {
        errF = err
        return
    }
}
if err := n.InvokeFunc(f); err != nil {
    return err
}
if errF != nil {
    return err
}
```

Reply:
Ok, done.